stdlib/public/core/ArrayCast.swift

```diff
@@ -152,7 +156,7 @@ internal func _arrayConditionalBridgeElements<SourceElement, TargetElement>(
   let buf = _ContiguousArrayBuffer<TargetElement>(
     uninitializedCount: source.count, minimumCapacity: 0)
   
-  var p = buf.firstElementAddress
+  var p = buf.firstElementAddress!
```

Comment:
Dmitri, Dave, and I talked about this offline and we decided I should make `firstElementAddress` non-optional, replacing any potentially-nil pointers with the base address of the shared empty array.